ext/RootsForwardDiffExt.jl

```diff
@@ -0,0 +1,39 @@
+module RootsForwardDiffExt
+
+using Roots
+import Roots: solve
+using ForwardDiff
+import ForwardDiff: Dual, derivative, gradient, partials
+
+
+function solve(ZP::ZeroProblem, M::Roots.AbstractUnivariateZeroMethod, p::Dual{T}; kwargs...) where {T}
+
+    f = ZP.F
+    pᵥ = p.value
+
+    xᵅ = solve(ZP, M, pᵥ; kwargs...)
+
+    fₓ = derivative(x -> f(x, pᵥ), xᵅ)
+    fₚ = derivative(p -> f(xᵅ, p), pᵥ)
+
+    Dual{T}(xᵅ, - fₚ / fₓ)
+end
+
+function solve(ZP::ZeroProblem,
+               M::Roots.AbstractUnivariateZeroMethod,
+               p::AbstractVector{<:Dual{Z,T,N}}; kwargs...) where {Z,T<:Real,N}
```

Comment:
This could be generalized to
```suggestion
               p::AbstractArray{<:Dual{T,<:Real}}; kwargs...) where {T}
```